games/ultimate_tic_tac_toe/game.py

```diff
@@ -0,0 +1,73 @@
+from base.game import Game
+from games.ultimate_tic_tac_toe.state import UTTTState
+from games.ultimate_tic_tac_toe.action import UTTTAction
+from typing import List, Optional
+import numpy as np
+
+
+class UltimateTicTacToeGame(Game[UTTTState, UTTTAction]):
+    def __init__(self, player_sign='X', opponent_sign='O'):
+        self.player_sign = player_sign
+        self.opponent_sign = opponent_sign
+        initial_state = self._initial_game_state()
+        super().__init__(initial_state)
+
+    def _initial_game_state(self) -> UTTTState:
+        return UTTTState(board=np.full((9, 3, 3), ' '), curr_block=4)
+
+    def actions_for(self, state: UTTTState, is_opponent: bool) -> List[UTTTAction]:
+        sign = self._get_sign(is_opponent)
+        if ' ' not in state.board[state.curr_block] or self._is_block_terminated(state, state.curr_block):
+            return [UTTTAction(sign, block, row, col) for block in range(9) for row in range(3) for col in range(3)
+                    if state.board[block, row, col] == ' ' and not self._is_block_terminated(state, block)]
+        else:
+            return [UTTTAction(sign, state.curr_block, row, col) for row in range(3) for col in range(3)
+                    if state.board[state.curr_block, row, col] == ' ']
+
+    def take_action(self, state: UTTTState, action: UTTTAction) -> UTTTState:
+        return action.apply(state)
+
+    def _value_for_terminal(self, state: UTTTState) -> float:
+        winner = self._find_winner(state)
+        if winner == self.player_sign:
+            return 1
+        if winner == self.opponent_sign:
+            return -1
+        return 0
+
+    def is_terminal_state(self, state: UTTTState) -> bool:
+        board = np.reshape([self._check_3x3(state.board[block]) for block in range(9)], (3, 3))
```

Comment:
Z tego co widzę to powtarzasz to dwa razy. 
Może lepiej nazwać akcję `cumulate` i wynieść do osobnej funkcji?